projects/common/src/time/time-range.service.test.ts

```diff
@@ -1,13 +1,40 @@
 import { ActivatedRoute, convertToParamMap } from '@angular/router';
+import {
+  NavigationService,
+  RelativeTimeRange,
+  TimeDuration,
+  TimeRange,
+  TimeRangeService,
+  TimeUnit
+} from '@hypertrace/common';
 import { runFakeRxjs } from '@hypertrace/test-utils';
 import { createServiceFactory, mockProvider } from '@ngneat/spectator/jest';
 import { NEVER, Observable, of } from 'rxjs';
 import { map } from 'rxjs/operators';
-import { NavigationService } from '../navigation/navigation.service';
 import { FixedTimeRange } from './fixed-time-range';
-import { TimeRangeService } from './time-range.service';
+
+interface NavMock {
+  queryParamTime: string | undefined;
+  defaultTimeRange: TimeRange | undefined;
+}
 
 describe('Time range service', () => {
+  let relativeTimeRange = new RelativeTimeRange(new TimeDuration(1, TimeUnit.Hour));
+  const firstMockRoute: NavMock = {
+    queryParamTime: '1573255100253-1573255111159',
+    defaultTimeRange: relativeTimeRange
+  };
+  const secondMockRoute: NavMock = {
+    queryParamTime: undefined,
+    defaultTimeRange: relativeTimeRange
+  };
+  const buildMockRoute = (navMock: NavMock): ActivatedRoute =>
+    // tslint:disable-next-line: no-object-literal-type-assertion
+    (({
+      queryParamMap: of(convertToParamMap({ time: navMock.queryParamTime })),
+      snapshot: { data: { defaultTimeRange: navMock.defaultTimeRange } }
+    } as unknown) as ActivatedRoute);
+
```

Comment:
pass featureResolverService implementation (many examples in test code) and test both scenarios

Reply:
not required anymore with the discussed refactor which doesn't involve changing this service, this file to be reverted back